corehq/apps/reports/standard/cases/case_data.py

```diff
@@ -555,6 +559,209 @@ def close_case_view(request, domain, case_id):
     return HttpResponseRedirect(reverse('case_data', args=[domain, case_id]))
 
 
+MAX_CASE_COUNT = 10
+MAX_SUBCASE_DEPTH = 3
+
+
+@location_safe
+def get_cases_and_forms_for_deletion(request, domain, case_id):
+    delete_cases = []  # list of cases to be soft deleted
+    delete_forms = []  # list of forms to be soft deleted
+
+    # For formatting the list of cases/submission forms
+    cases = {}  # structured like: {case: {form: {case touched by form: actions taken by form}}}
+    case_names = {}  # {case_id: case_name}
+    form_names = {}  # {form_ids: form name}
+    reopened_cases = {}
+    affected_cases = {}
+
+    update_actions = [const.CASE_ACTION_INDEX,
+                      const.CASE_ACTION_UPDATE,
+                      const.CASE_ACTION_ATTACHMENT,
+                      const.CASE_ACTION_COMMTRACK,
+                      const.CASE_ACTION_REBUILD]
+
+    def get_forms_for_deletion_from_case(case, subcase_count):
+        delete_cases.append(case.case_id)
+        if len(delete_cases) > MAX_CASE_COUNT or subcase_count >= MAX_SUBCASE_DEPTH:
```

Comment:
Is it true that some cases will be impossible to delete because of these restrictions? Example:
- A case that was touched by a case upload, which I believe batches up to 100 cases into each form submission.
- Any case with more than 2 levels of subcases.

Reply:
As of right now, yes, the case in the first example can’t be deleted and neither can any other case that were uploaded in the same batch. I’m open to changing the value of `MAX_CASE_COUNT` though, since 10 was chosen arbitrarily. For the second example though, the case can’t be deleted but you can go into the subcases and delete those first and come back to the original case after that. I’m also open to changing the value of `MAX_SUBCASE_DEPTH`.

Reply:
@millerdev do you know where we batch the 100 cases in case importer? It would be nice to make this a settings variable that can also be accessed by this case deletion code. I'm not too familiar with case importer, so couldn't find it quickly...

Reply:
so should `MAX_CASE_COUNT` be set to equal this batch size?